pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -178,11 +179,17 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
-	if err = utils.CheckLicenseKey(ctx, r.client); err != nil {
+	license, err := utils.FetchLicenseKey(ctx, r.client)
+	if err != nil {
 		r.status.SetDegraded("License not found", err.Error())
 		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
+	if !utils.IsFeatureActive(license, common.ThreatDefenseFeature) {
+		log.Info("IntrusionDetection is not activated as part of this license")
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
Same here, if the CR is provided but the license doesn't enable it, we should mark the user's intent as degraded.

Reply:
@caseydavenport I added the change

```
kubectl get tigerastatus compliance -o yaml 

apiVersion: operator.tigera.io/v1
kind: TigeraStatus
metadata:
  creationTimestamp: "2021-02-10T19:16:11Z"
  generation: 1
  managedFields:
  - apiVersion: operator.tigera.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:spec: {}
      f:status:
        .: {}
        f:conditions: {}
    manager: operator
    operation: Update
    time: "2021-02-10T19:16:11Z"
  name: compliance
  resourceVersion: "197120"
  selfLink: /apis/operator.tigera.io/v1/tigerastatuses/compliance
  uid: e140454f-1e28-4d7f-a3b0-61dc08cd5d9a
spec: {}
status:
  conditions:
  - lastTransitionTime: "2021-02-10T19:16:11Z"
    message: License does not support this feature
    reason: Feature is not active
    status: "True"
    type: Degraded
  - lastTransitionTime: "2021-02-10T19:16:16Z"
    status: "False"
    type: Available
  - lastTransitionTime: "2021-02-10T19:16:21Z"
    status: "False"
    type: Progressing

```